jenkins/databricks/build.sh

```diff
@@ -15,437 +15,292 @@
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$ORC_MAPREDUCE_JAR \
-   -DgroupId=org.apache.orc \
-   -DartifactId=orc-mapreduce \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B install:install-file \
-   -Dmaven.repo.local=$M2DIR \
-   -Dfile=$JARDIR/$PROTOBUF_JAR \
-   -DgroupId=com.google.protobuf \
-   -DartifactId=protobuf-java \
-   -Dversion=$SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS \
-   -Dpackaging=jar
-
-$MVN_CMD -B -Ddatabricks -Dbuildver=$BUILDVER clean package -DskipTests $MVN_OPT
-
+## can be overwritten by shell variables, e.g. "BASE_SPARK_VERSION=3.1.2 MVN_OPT=bash build.sh"
+
+# default SPARK Version is extracted from DB Spark environment
+DEFAULT_SPARK_BASE_VERSION=$(< /databricks/spark/VERSION)
+
+# list of supported sparks
+SUPPORTED_SPARKS=("3.1.2" "3.2.1" "3.3.0")
+
+# Map of software versions for each dependency.
+declare -A sw_versions
+# Map of jar file locations of all dependencies
+declare -A dep_jars
+# Map of string arrays to hold the groupId and the artifactId for each JAR
+declare -A artifacts
+
+# Initializes the scripts and the variables based on teh arguments passed to the script.
+initialize()
+{
+    # install rsync to be used for copying onto the databricks nodes
+    sudo apt install -y maven rsync
+
+    # Archive file location of the plugin repository
+    SPARKSRCTGZ=${SPARKSRCTGZ:-''}
+
+    # version of Apache Spark we are building against
+    BASE_SPARK_VERSION=${BASE_SPARK_VERSION:-${DEFAULT_SPARK_BASE_VERSION}}
+
+    # check early that the that the spark base version is supported
+    if [[ !  ${SUPPORTED_SPARKS[*]}  =~  ${BASE_SPARK_VERSION}  ]]; then
+        echo "${BASE_SPARK_VERSION} Unexpected Spark version. Currently, ${SUPPORTED_SPARKS[*]} are only supported."
+        exit 1
+    fi
+
+    BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS=${BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS:-$BASE_SPARK_VERSION}
+    ## '-Pfoo=1,-Dbar=2,...' to '-Pfoo=1 -Dbar=2 ...'
+    MVN_OPT=${MVN_OPT//','/' '}
+    BUILDVER=$(echo ${BASE_SPARK_VERSION} | sed 's/\.//g')db
+    # the version of Spark used when we install the Databricks jars in .m2
+    BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS=${BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS:-$BASE_SPARK_VERSION}
+    SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS=$BASE_SPARK_VERSION_TO_INSTALL_DATABRICKS_JARS-databricks
+
+    # pull normal Spark artifacts and ignore errors then install databricks jars, then build again.
+    # this should match the databricks init script.
+    JARDIR=/databricks/jars
+
+    if [[ -n $SPARKSRCTGZ ]]
+    then
+        rm -rf spark-rapids
+        mkdir spark-rapids
+        echo  "tar -zxf $SPARKSRCTGZ -C spark-rapids"
+        tar -zxf $SPARKSRCTGZ -C spark-rapids
+        cd spark-rapids
+    fi
+
+    # Now, we can set the WORKSPACE
+    export WORKSPACE=$(pwd)
```

Comment:
nit it's already in the environment
```suggestion
    export WORKSPACE=$PWD
```